js/data/bucket.js

```diff
@@ -150,34 +147,42 @@ Bucket.prototype.createArrays = function() {
         var programInterface = this.programInterfaces[programName];
 
         if (programInterface.vertexBuffer) {
-            var vertexAddMethodName = this.getAddMethodName(programName, 'vertex');
             var vertexBufferName = this.getBufferName(programName, 'vertex');
 
             var VertexArrayType = new StructArrayType({
-                members: this.attributes[programName].enabled,
+                members: this.attributes[programName].coreAttributes,
```

Comment:
Could these be called `layoutAttributes`? Then we'd have `layoutAttributes` as a function of layout properties and `paintAttributes` as a function of paint properties.